src/structures/paging/phys_mem_alloc.rs

```diff
@@ -0,0 +1,11 @@
+
+use structures::paging::{PageSize, PhysFrame};
+
+/// An allocator of physical pages of memory.
+pub trait PhysMemAllocator {
+    /// Allocate a physical frame of size `S` if possible.
+    fn alloc<S: PageSize>(&mut self) -> Option<PhysFrame>;
```

Comment:
The generic `S` parameter is not used. I think you meant to use `PhysFrame<S>` as return type.